modules/statistics/statistics.js

```diff
@@ -710,16 +706,26 @@ Statistics.reportGlobalError = function(error) {
 };
 
 /**
- * Sends event to analytics and callstats.
- * @param {string} eventName the event name.
- * @param {Object} data the data to be sent.
+ * Sends event to analytics and logs a message to callstats.
+ *
+ * @param {string | Object} eventName the event name, or an object which
+ * represents the entire event.
+ * @param {Object} properties properties to attach to the event
  */
-Statistics.sendEventToAll = function(eventName, data) {
-    this.analytics.sendEvent(eventName, data);
-    Statistics.sendLog(
-        JSON.stringify(
-            {
-                name: eventName,
-                data
-            }));
+Statistics.sendEventAndLog = function(eventName, properties = {}) {
+    let eventToLog = null;
+
+    // Also support an API with a single object as an event.
+    if (typeof eventName === 'object') {
```

Comment:
While we're on the subject of `null`, `typeof null` does equal `'object'`. Is `null` a supposed value?

Reply:
No, I don't want to accept "null" as an event name (or event object). I've added "if(!eventName) return;" to the top of the function